### PR TITLE
record tos consent for steemit

### LINKED ISSUE #232: must record user consent to TOS
As a user who signed up via faucet, it is important to steemit that I have agreed to the current version of the TOS before my account is created and that that fact is recorded in the `conveyor` tagging database.

This means the current Steemit TOS needs to be displayed to the user during account creation, and they must agree to it, and then that agreement and TOS version must be recorded via setting a tag on that user via `conveyor`.

### PR DESCRIPTION
**Pull request by bnchdrff**
closes #232 

requires #365 

screenshot from 2018-06-08 13-02-59

### CODE REVIEW

File: src/components/Form/CreateAccount/Password.js [hunk 1]
```diff
@@ -52,6 +54,12 @@ class Password extends React.Component {
         }
     };
 
+    requireTerms = (rule, value, callback) => {
+        if (value) callback();
+
+        callback(false);
```
Comment (jnordberg) [inline src/components/Form/CreateAccount/Password.js]: callback called twice here

File: src/reducers/user.js [hunk 2]
```diff
@@ -15,14 +15,20 @@ const SET_PREFIX = 'user/SET_PREFIX';
 const SET_COMPLETED = 'user/SET_COMPLETED';
 const SET_TRACKING_ID = 'user/SET_TRACKING_ID';
 
+// Provide the referrer based on whatever our ref is set to at page load.
+const referrerMatch = window.location.search.match(/\?ref=([a-zA-Z*])/);
```
Comment (jnordberg) [inline src/reducers/user.js]: wasn't this added in another pr?

Comment (bnchdrff) [reply src/reducers/user.js]: yes; this PR is reliant on those changes and they're included in this branch in commit https://github.com/steemit/faucet/pull/366/commits/8466046f5729bcec77997fb626dd3e63a926c623

File: src/components/Form/CreateAccount/Password.js (hunk 1 above)
Comment (bnchdrff) [reply src/components/Form/CreateAccount/Password.js]: oof yup.

**Pull request merged**